dbt-bigquery/src/dbt/adapters/bigquery/clients.py

```diff
@@ -56,12 +56,16 @@ def create_dataproc_batch_controller_client(
 
 @BQ_DEFAULT_RETRY
 def _create_bigquery_client(credentials: BigQueryCredentials) -> BigQueryClient:
+    kwargs_options = {}
+    if credentials.api_endpoint:
+        kwargs_options["api_endpoint"] = credentials.api_endpoint
+
     return BigQueryClient(
         credentials.execution_project,
         create_google_credentials(credentials),
         location=getattr(credentials, "location", None),
         client_info=ClientInfo(user_agent=f"dbt-bigquery-{dbt_version.version}"),
-        client_options=ClientOptions(quota_project_id=credentials.quota_project),
+        client_options=ClientOptions(quota_project_id=credentials.quota_project, **kwargs_options),
```

Comment:
```suggestion
        client_options=ClientOptions(quota_project_id=credentials.quota_project, api_endpoint=credentials.api_endpoint),
```
api_endpoint defaults to None so should be safe to just pass this directly and then we don't need the `kwargs_options`